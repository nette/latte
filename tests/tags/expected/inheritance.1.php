<?php
%A%
final class Template%a% extends Latte\Runtime\Template
{
	public const Blocks = [
		['title' => 'blockTitle', 'content' => 'blockContent'],
	];


	public function main(array $ʟ_args): void
	{
%A%
		$this->createTemplate('inc', $this->params, 'includeblock')->renderToContentType('html') /* line %d% */;
		echo "\n";
		$this->renderBlock('title', get_defined_vars()) /* line %d% */;
		echo '

';
		$this->renderBlock('content', get_defined_vars()) /* line %d% */;
	}


	public function prepare(): array
	{
		extract($this->params);

		if (!$this->getReferringTemplate() || $this->getReferenceType() === 'extends') {
			foreach (array_intersect_key(['person' => '11'], $this->params) as $ʟ_v => $ʟ_l) {
				trigger_error("Variable \$$ʟ_v overwritten in foreach on line $ʟ_l");
			}
		}
		$this->parentName = 'parent';
		$this->createTemplate('inc', $this->params, "import")->render() /* line %d% */;
		return get_defined_vars();
	}


	/** {block title} on line %d% */
	public function blockTitle(array $ʟ_args): void
	{
		extract($this->params);
		extract($ʟ_args);
		unset($ʟ_args);

		echo 'Homepage | ';
		$this->renderBlockParent('title', get_defined_vars()) /* line %d% */;
		$this->renderBlockParent('title', get_defined_vars()) /* line %d% */;
	}


	/** {block content} on line %d% */
	public function blockContent(array $ʟ_args): void
	{
		extract($this->params);
		extract($ʟ_args);
		unset($ʟ_args);

		echo '	<ul>
';
		foreach ($people as $person) /* line %d% */ {
			echo '		<li>';
			echo LR\Filters::escapeHtmlText($person) /* line %d% */;
			echo '</li>
';

		}

		echo '	</ul>
	Parent: ';
		echo LR\Filters::escapeHtmlText(gettype($this->getReferringTemplate())) /* line %d% */;
		echo "\n";
	}
}
