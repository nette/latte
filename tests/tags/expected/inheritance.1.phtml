<?php
%A%
final class Template%a% extends Latte\Runtime\Template
{
	protected const BLOCKS = [
		['title' => 'blockTitle', 'content' => 'blockContent'],
	];


	public function main(): array
	{
%A%
		$this->createTemplate("inc", $this->params, 'includeblock')->renderToContentType('html') /* line %d% */;
		echo "\n";
		if ($this->getParentName()) {
			return get_defined_vars();
		}
		$this->renderBlock('title', get_defined_vars()) /* line %d% */;
		echo '

';
		$this->renderBlock('content', get_defined_vars()) /* line %d% */;
%A%
	}


	public function prepare(): void
	{
		extract($this->params);
		$this->parentName = "parent";
		$this->createTemplate("inc", $this->params, "import")->render() /* line %d% */;
		if (!$this->getReferringTemplate() || $this->getReferenceType() === "extends") {
			foreach (array_intersect_key(['person' => '11'], $this->params) as $ʟ_v => $ʟ_l) {
				trigger_error("Variable \$$ʟ_v overwritten in foreach on line $ʟ_l");
			}
		}
	}


	/** {block title} on line %d% */
	public function blockTitle(array $ʟ_args): void
	{
		extract($this->params);
		extract($ʟ_args);
		unset($ʟ_args);

		echo 'Homepage | ';
		$this->renderBlockParent('title', get_defined_vars()) /* line %d% */;
		$this->renderBlockParent('title', get_defined_vars()) /* line %d% */;
	}


	/** {block content} on line %d% */
	public function blockContent(array $ʟ_args): void
	{
		extract($this->params);
		extract($ʟ_args);
		unset($ʟ_args);

		echo '	<ul>
';
		$iterations = 0;
		foreach ($people as $person) /* line %d% */ {
			echo '		<li>';
			echo LR\Filters::escapeHtmlText($person) /* line %d% */;
			echo '</li>
';
			$iterations++;
		}

		echo '	</ul>
	Parent: ';
		echo LR\Filters::escapeHtmlText(gettype($this->getReferringTemplate())) /* line %d% */;
		echo "\n";
	}

}
