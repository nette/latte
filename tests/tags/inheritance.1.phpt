<?php

/**
 * Test: {extends ...} test I.
 */

declare(strict_types=1);

use Tester\Assert;

require __DIR__ . '/../bootstrap.php';


$latte = new Latte\Engine;
$latte->setLoader(new Latte\Loaders\StringLoader([
	'parent' => file_get_contents(__DIR__ . '/templates/parent.latte'),

	'main' => <<<'XX'

		{extends "parent"}

		{import "inc"}
		{include file "inc" with blocks}

		{block title}Homepage | {include parent}{include parent}{/block}

		{block content}
			<ul>
			{foreach $people as $person}
				<li>{$person}</li>
			{/foreach}
			</ul>
			Parent: {gettype($this->getReferringTemplate())}
		{/block}

		XX,

	'inc' => '{define test /}',
]));

Assert::matchFile(
	__DIR__ . '/expected/inheritance.1.php',
	$latte->compile('main'),
);
Assert::matchFile(
	__DIR__ . '/expected/inheritance.1.html',
	$latte->renderToString('main', ['people' => ['John', 'Mary', 'Paul']]),
);
Assert::matchFile(
	__DIR__ . '/expected/inheritance.1.parent.php',
	$latte->compile('parent'),
);
