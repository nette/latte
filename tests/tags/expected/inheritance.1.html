<!DOCTYPE html>
<head>
	<title>HOMEPAGE | MY WEBSITEMY WEBSITE</title>
</head>

<body>
	<div id="sidebar">
		<ul>
			<li><a href="/">Homepage</a></li>
		</ul>
	</div>

	<div id="content">
	<ul>
		<li>John</li>
		<li>Mary</li>
		<li>Paul</li>
	</ul>
	Parent: NULL


		JOHN
		MARY
		PAUL

	PARENT: NULL
	</div>
</body>
</html>
Parent: main/extends
